pages/11ty/markdown.shortcut.js

```diff
@@ -71,6 +77,53 @@ class MDRenderer {
     }
     return github.srcUrl + linkPath;
   }
+
+  static generateHeadersIds(content) {
+    const headerTags = ['h1', 'h2', 'h3', 'h4'];
+    const idLengthLimit = 20;
+
+    headerTags.forEach(tag => {
+      const headers = content.getElementsByTagName(tag);
+
+      for (let header of headers) {
+        const text = header.textContent;
+        const id = text
+          .replace(/(?:^\w|[A-Z]|\b\w)/g, (word, index) => (index === 0 ? word.toLowerCase() : word.toUpperCase()))
+          .replace(/[^\w\s]|_/g, "").replace(/\s+/g, "")
+          .substring(0, idLengthLimit);
+
+        header.setAttribute('id', id);
+
+        this.generateAnchors(content, text, `#${id}`)
+      }
+    });
+    return content;
+  }
+
+  static generateGloballyDefinedTermsLinks (content) {
+    const globallyDefinedTerms = {
+      ESL_Base_Element: 'https://esl-ui.com/core/esl-base-element/',
+      ESL_Mixin_Element: 'https://esl-ui.com/core/esl-mixin-element/'
+      // Add other globally defined terms as needed
+    };
+
+    Object.keys(globallyDefinedTerms).forEach(term => {
+      this.generateAnchors(content, term.replace(/_/g, " "), globallyDefinedTerms[term]);
+    });
+  }
+
+  static generateAnchors(content, text, link) {
```

Comment:
The current process makes the same things (and not the light ones) a couple of times.
Propose to split it a little bit...

We can use the following snippet to find all the text nodes in DOM.
```
function findTextNodes(root: Node): Node[] {
  let all: Node[] = [];
  for (let node = root.firstChild; node; node = node.nextSibling) {
    if (node.nodeType === 3) all.push(node);
    else all = all.concat(findTextNodes(node));
  }
  return all;
}
```

Then we have a simple and straight process:
```
 const nodes = findTextNodes(consent)
   .filter((node) => !node.parentElement.closest('h1, h2, h3, h4, h5, h6'));
 
 for (const [text, link] = Object.entries(terms)) {
    const url = resolveLink(link);
    const regex = new RegExp(`(^|\\s)${text}(\\s|[,\\.])`, 'g');
    for (const node of nodes) {
       node.textContent = node.textContent.replace(regex, `$1<a href="${url}">${text}</a>$2`);
    }
 }
```